packages/core/src/types/authMachine.ts

```diff
@@ -30,15 +30,18 @@ export interface SignInContext extends BaseFormContext {
 
 export interface SignUpContext extends BaseFormContext {
   login_mechanisms?: string[];
+  unverifiedAttributes?: Record<string, string>;
 }
 
 export interface ResetPasswordContext extends BaseFormContext {
   username?: string;
+  unverifiedAttributes?: Record<string, string>;
```

Comment:
Ah, so this is the same problem as https://github.com/aws-amplify/amplify-ui/pull/310#discussion_r698049403. I dived into this a little bit yesterday and found some clue I wrote in https://github.com/aws-amplify/amplify-ui/pull/316#issuecomment-909509741. tldr, Vue has problem inferring computed property types. (Not sure if it's a problem with Vue or rtp2...)
 
Ideally we would want to fix this error, but I'm good to descope this to another pr. We can revisit this later after @eddiekeller's state machine improvements.

Reply:
@wlee221 Thanks for doing the research! Yes, that must be what's happening, weird.